File: data-structure/2학년때/List/SinglyLinkedList/Student.java
package DataStructure.SinglyLinkedList;

class Student
{
    private String name;
    private int id;
    private ClubList club;

    Student(int id, String name)
    {
        this.id = id;
        this.name = name;
    }

    Student(int id, String name, ClubList club)
    {
        this.id = id;
        this.name = name;
        this.club = club;
    }

    void setId(int id) { this.id = id; }
    void setName(String name) { this.name = name; }
    void setClub(ClubList club) { this.club = club; }
    int getId() { return id; }
    String getName() { return name; }
    ClubList getClub() { return club; }

    public String toString()
    {
        if(club==null)
            return id+" "+name;
        else
            return id+" "+name+" "+club;
    }

}

File: 2-2 c++programming/발전기금관리프로그램/Student.cpp
#include "Student.h"

Student::Student(string id, string name, string department, string phone, int deposit)
	: Person(name, phone, deposit), studentID(id), department(department) {}

Student::~Student() {}

void Student::print() const
{
	cout << "[ 학생 ] " << getName() << "(학번:" << getID() << ", 학과:"
		<< getDepartment() << ") " << getPhone() << " " << getDeposit() << endl;
}

string Student::toString() const
{
	return "학생 " + getID() + " " + getName() + " " + getDepartment() + " "
		+ getPhone() + " " + to_string(getDeposit()) + "\n";
}

void Student::setID(string id)
{
	studentID = id;
}

void Student::setDepartment(string department)
{
	this->department = department;
}


string Student::getID() const
{
	return studentID;
}

string Student::getDepartment() const
{
	return department;
}